extensions/standard-processors/processors/AttributesToJSON.cpp

```diff
@@ -0,0 +1,174 @@
+                      "Writing to flowfile content will overwrite any existing flowfile content.")
+    ->isRequired(true)
+    ->withDefaultValue<std::string>(toString(WriteDestination::FLOWFILE_ATTRIBUTE))
+    ->withAllowableValues<std::string>(WriteDestination::values())
+    ->build());
+
+const core::Property AttributesToJSON::IncludeCoreAttributes(
+  core::PropertyBuilder::createProperty("Include Core Attributes")
+    ->withDescription("Determines if the FlowFile core attributes which are contained in every FlowFile should be included in the final JSON value generated.")
+    ->isRequired(true)
+    ->withDefaultValue<bool>(true)
+    ->build());
+
+const core::Property AttributesToJSON::NullValue(
+  core::PropertyBuilder::createProperty("Null Value")
+    ->withDescription("If true a non existing or empty attribute will be NULL in the resulting JSON. If false an empty string will be placed in the JSON.")
+    ->isRequired(true)
+    ->withDefaultValue<bool>(false)
+    ->build());
+
+core::Relationship AttributesToJSON::Success("success", "All FlowFiles received are routed to success");
+
+void AttributesToJSON::initialize() {
+  setSupportedProperties({
+    AttributesList,
+    AttributesRegularExpression,
+    Destination,
+    IncludeCoreAttributes,
+    NullValue
+  });
+  setSupportedRelationships({Success});
+}
+
+void AttributesToJSON::onSchedule(core::ProcessContext* context, core::ProcessSessionFactory* /*sessionFactory*/) {
+  std::string value;
+  if (context->getProperty(AttributesList.getName(), value) && !value.empty()) {
+    attribute_list_ = utils::StringUtils::splitAndTrimRemovingEmpty(value, ",");
+  }
+  if (context->getProperty(AttributesRegularExpression.getName(), value) && !value.empty()) {
+    attributes_regular_expression_ = std::regex(value);
+  }
+  write_destination_ = WriteDestination::parse(utils::parsePropertyWithAllowableValuesOrThrow(*context, Destination.getName(), WriteDestination::values()).c_str());
+  context->getProperty(IncludeCoreAttributes.getName(), include_core_attributes_);
+  context->getProperty(NullValue.getName(), null_value_);
+}
+
+bool AttributesToJSON::isCoreAttributeToBeFiltered(const std::string& attribute) const {
+  return !include_core_attributes_ && core_attributes_.find(attribute) != core_attributes_.end();
+}
+
+std::unordered_set<std::string> AttributesToJSON::getAttributesToBeWritten(const std::map<std::string, std::string>& flowfile_attributes) const {
+  std::unordered_set<std::string> attributes;
+
+  for (const auto& attribute : attribute_list_) {
+    if (!isCoreAttributeToBeFiltered(attribute)) {
+      attributes.insert(attribute);
+    }
+  }
+
+  if (attributes_regular_expression_) {
+    for (const auto& [key, value] : flowfile_attributes) {
+      if (!isCoreAttributeToBeFiltered(key) && std::regex_search(key, attributes_regular_expression_.value())) {
```

Comment:
I think NiFi matches the regex against the whole attribute name, not only part of if.  I would change `regex_search` to `regex_match` so we do the same.

Reply:
You are right, updated the code and the tests in dc9f6aa438e986a8ae6d1bacb85ca4fc677cb175